src/core/core.scale.js

```diff
@@ -553,8 +566,13 @@ var Scale = Element.extend({
 
 		me.handleMargins();
 
-		me.width = minSize.width;
-		me.height = minSize.height;
+		if (isHorizontal) {
+			me.width = chart.width - me.margins.left - me.margins.right;
```

Comment:
Why is this needed?

Reply:
To make it the same as the `chartArea.width`: https://github.com/chartjs/Chart.js/blob/0d3754254ab3293bb68f57023d47ba53d87fe483/src/core/core.layouts.js#L94

`placeBoxes` will update the width to match `chartArea.width`: https://github.com/chartjs/Chart.js/blob/0d3754254ab3293bb68f57023d47ba53d87fe483/src/core/core.layouts.js#L179

We need the width here to equal the eventual `width` because `_autoSkip` uses it to calculate how many labels it can show. Before it was being set to a wrong value

Reply:
Right. We should really have a place for stuff after `placeBoxes`. I guess `labelRotation` is also currently computed using wrong width.

Reply:
`calculateTickRotation` actually uses the right width: https://github.com/chartjs/Chart.js/blob/0d3754254ab3293bb68f57023d47ba53d87fe483/src/core/core.scale.js#L431

Would it seem better to you if I did this:
```
me.width = minSize.width = Math.min(minSize.width, chart.width - me.margins.left - me.margins.right);
me.height = minSize.height = Math.min(minSize.height, chart.height - me.margins.top - me.margins.bottom);
```

It seems fine to me that we should be able to constrain the width not to be larger than chart size minus margins

Reply:
I think this works fine. I just don't like that we calculate stuff from chart dimensions in a scale.